src/librustc_borrowck/borrowck/mod.rs

```diff
@@ -68,7 +68,7 @@ impl<'a, 'tcx> Visitor<'tcx> for BorrowckCtxt<'a, 'tcx> {
     }
 
     fn visit_fn(&mut self, fk: FnKind<'tcx>, fd: &'tcx hir::FnDecl,
-                b: hir::ExprId, s: Span, id: ast::NodeId) {
+                b: hir::BodyId, s: Span, id: ast::NodeId) {
```

Comment:
It should not because `Foo` is not `Copy`, and we are moving it from a `&Foo` -- e.g., this does not compile: https://is.gd/zU2BVo